### PR TITLE
cmake: allow llvm without zstd

### PR DESCRIPTION
**Pull request by junghans**
For https://github.com/lfortran/lfortran/issues/3044#issuecomment-1901247501

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -135,7 +135,10 @@ set(WITH_TARGET_AARCH64 no CACHE BOOL "Enable target AARCH64")
 set(WITH_TARGET_X86 no CACHE BOOL "Enable target X86")
 set(WITH_TARGET_WASM no CACHE BOOL "Enable target WebAssembly")
 if (WITH_LLVM)
-    find_package(StaticZSTD REQUIRED)
+    set(WITH_ZSTD yes CACHE BOOL "Detect (and required static libzstd for llvm")
```
Comment (certik) [inline CMakeLists.txt:138]: Did you mean to write this:
```suggestion
    set(WITH_ZSTD yes CACHE BOOL "Detect (and require) static libzstd for llvm")
```

**Review by certik — APPROVED** [review]
I think that's fine.

**Comment by certik** [comment]
Thanks!

**Comment by junghans** [comment]
Another release would be good at some point, too!

**Comment by certik** [comment]
Done: https://github.com/lfortran/lfortran/releases/tag/v0.31.0, the release tarball will be uploaded automatically soon.

**Pull request merged**